File: cbits/IloEnv.cpp
#include "../include/IloEnv.h"

IloEnv* new_iloEnv(void){
    return (new IloEnv());
}

void delete_iloEnv(IloEnv* e){
    delete e;
    std::cout << "===========+> deleting env" << std::endl;
    return;
}

File: src/IloCplex/IloObject/IloEnv.hs
{-# LANGUAGE ForeignFunctionInterface #-}
module IloCplex.IloObject.IloEnv where

import Foreign.Ptr
import Foreign.ForeignPtr

foreign import ccall "include/IloEnv.h new_iloEnv"
    new_iloEnv :: IO (Ptr ())
foreign import ccall "include/IloEnv.h &delete_iloEnv"
    delete_iloEnv :: FunPtr(Ptr () -> IO ())


newtype IloEnv = IloEnv (ForeignPtr ())

newIloEnv :: IO IloEnv
newIloEnv = IloEnv <$> (new_iloEnv >>= newForeignPtr delete_iloEnv)

